crates/rethnet_evm/benches/state_history.rs

```diff
@@ -0,0 +1,100 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+use rethnet_eth::B256;
+
+mod util;
+use util::{bench_sync_state_method, prep_no_op, SNAPSHOT_SCALES};
+
+fn bench_checkpoint(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateHistory::checkpoint()",
+        prep_no_op,
+        |mut state, _number_of_accounts| {
+            let result = state.checkpoint();
+            debug_assert!(result.is_ok());
+        },
+        &[0],
+        &[0],
+    );
+}
+
+fn bench_make_snapshot(c: &mut Criterion) {
+    bench_sync_state_method(
+        c,
+        "StateHistory::make_snapshot",
+        prep_no_op,
+        |mut state, _number_of_accounts| {
+            state.make_snapshot();
+        },
+        &[0],
+        &SNAPSHOT_SCALES,
+    );
+}
+
+fn bench_remove_snapshot(c: &mut Criterion) {
+    let snapshot: std::cell::RefCell<B256> = Default::default();
+    bench_sync_state_method(
+        c,
+        "StateHistory::remove_snapshot",
+        |state, _| {
+            snapshot
+                .borrow_mut()
+                .assign_from_slice(state.make_snapshot().as_bytes());
+        },
+        |mut state, _number_of_accounts| {
+            state.remove_snapshot(&snapshot.borrow());
+        },
+        &[0],
+        &SNAPSHOT_SCALES,
```

Comment:
Same as `bench_make_snapshot`

Reply:
addressed in https://github.com/NomicFoundation/hardhat/pull/3921/commits/a65aa9bcf22849445a0ef338ca9f2cc9a181afa4